colossalai/kernel/triton/rms_layernorm.py

```diff
@@ -66,4 +65,4 @@ def rms_layernorm(x, weight, eps, norm_output=None):
 
         # enqueue kernel
         _rmsnorm_kernel[(M,)](x, y, weight, x.stride(0), N, eps, BLOCK_SIZE=BLOCK_SIZE, num_warps=num_warps)
-        return y
+        return y, x
```

Comment:
Why do you need to return the input as input is not modified at all.

Reply:
In this way，we can avoid implicit DToD memcopy operation.

Reply:
What is the difference between these two cases?

- case 1:
```python
out = rms_layernorm(x, weight)

# do something on x
```

- case 2
```

```python
out, x = rms_layernorm(x, weight)

# do something on x
```
```

Reply:
I suspect that this is what actually happens.

Reply:
As `rms_layernorm` does not touch on `x` at all, the input argument should remain unchanged no matter if `x` is returned.

Reply:
fixed